utils/logger/logger_test.go

```diff
@@ -0,0 +1,348 @@
+// this is a new logger interface for mattermost
+
+package logger
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+type LogMessage struct {
+	Context map[string]string
+	File    string
+	Message string
+}
+
+// ensures that values can be recorded on a Context object, and that the data in question is serialized as a part of the log message
+func TestSerializeContext(t *testing.T) {
+	t.Run("Context values test", func(t *testing.T) {
+		ctx := context.Background()
+
+		expectedUserID := "some-fake-user-id"
+		ctx = WithUserID(ctx, expectedUserID)
+
+		expectedRequestID := "some-fake-request-id"
+		ctx = WithRequestID(ctx, expectedRequestID)
+
+		serialized := serializeContext(ctx)
+
+		userID, ok := serialized["user_id"]
+		if !ok {
+			t.Error("UserID was not serialized")
+		}
+		if userID != expectedUserID {
+			t.Errorf("UserID = %v, want %v", userID, expectedUserID)
+		}
+
+		requestID, ok := serialized["request_id"]
+		if !ok {
+			t.Error("RequestID was not serialized")
+		}
+		if requestID != expectedRequestID {
+			t.Errorf("RequestID = %v, want %v", requestID, expectedRequestID)
+		}
+	})
+}
+
+// ensures that an entire log message with an empty context can be properly serialized into a JSON object
+func TestSerializeLogMessageEmptyContext(t *testing.T) {
+	emptyContext := context.Background()
+
+	var logMessage = "This is a log message"
+	var serialized = serializeLogMessage(emptyContext, logMessage)
+
+	var deserialized LogMessage
+	json.Unmarshal([]byte(serialized), &deserialized)
+
+	if len(deserialized.Context) != 0 {
+		t.Error("Context is non-empty")
+	}
+	var expectedFileSuffix = "/platform/utils/logger/logger_test.go"
+	if !strings.HasSuffix(deserialized.File, expectedFileSuffix) {
```

Comment:
Well, you can get the current file path with `runtime.Caller(0)`. That will have a "platform/utils/logger/logger.go" suffix. So take the left of that and trim it from every path.